docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,122 @@
+---
+
+A work-in-progress program for permissionless management of stake pools that are
+only delegated to one validator.
+
+## Operation
+
+The single-validator manager program defines a canonical stake pool for any
+validator on the network, which can be created by anyone.
+
+The program manages the "delegation strategy" of the pool, which is to maximize
+the amount of stake on one validator, while maintaining a small amount of SOL
+in the reserve for small stakers.
+
+### Create Pool
+
+The `CreateStakePool` instruction allows anyone to create the canonical stake
+pool for a validator, at the cost of ~0.0124 SOL for account rent-exemption of
+the stake pool, validator list, reserve, mint, and fee account.
+
+After the pool is created, the reserve requires at least stake rent-exemption plus
+the minimum delegation amount to actually add the validator to the pool, which
+costs 1.0028288 SOL. The user receives pool tokens for this deposit.
+
+After the SOL deposit, the `AddValidatorToPool` instruction on the pool management
+program allows anyone to add the stake account for that validator.
+
+Once the added stake account is active, anyone may deposit stake accounts.
+
+### Depositing and withdrawing stake
+
+If stakers deposit or withdraw stake accounts from the pool, everything works
+seamlessly. Stakers receive pool tokens for their deposited stake account, and
+may use the pool tokens to withdraw a stake account.
+
+Activated stake accounts on the same validator are all fungible, so there's no
+other economics to worry about.
+
+### Depositing and withdrawing SOL
+
+As mentioned in the fees section, only SOL deposits have a fee, and everything
+else can be done freely.
+
+To maximize returns for pool depositors, the program allows anyone to maximally
+deploy the reserve to the validator, with important limits.
+
+Single-validator stake pools are an option for small stakers to earn inflation
+rewards by staking their SOL, but if stakers have less than the minimum delegation
+amount, they cannot withdraw stake accounts, so they need a way to withdraw SOL
+from the reserve.
+
+To give a way out for these users, the reserve aims to maintain the minimum delegation
+amount plus stake rent-exemption, or 1.0028288 SOL. If the reserve has less than
+this amount, any user can call `DecreaseValidatorStake` to decrease the minimum
+delegation amount plus rent-exemption.
+
+On the other hand, if the reserve receives enough SOL deposits equalling at least
+two times this amount, then anyone can call `IncreaseValidatorStake` on the manager
+program to activate everything in the reserve, minus the minimum delegation plus
+rent-exemption.
+
+### Griefing
+
+The permissionless increase instruction allows an attack to make the pool less
+performant.
+
+For example, at the start of the epoch, a malicious user can deposit enough SOL to get
+just above the limit for increasing, and immediate run the increase. This prevents
```

Comment:
The stake pool prevents it, since there's only one transient account allowed, either increasing or decreasing, once per epoch.

We can modify the stake pool program to make this case work, but it requires breaking the interface since you need to add an ephemeral account. On the flip side, it might be ok if it's optional and at the end...

It's how redelegate works, and it makes me realize that all of stake pools should have been done this way.

Reply:
ah i guess this comment makes more sense here...

https://github.com/solana-labs/solana-program-library/pull/3881#discussion_r1044684302